cmd/sync/sync.go

```diff
@@ -23,17 +27,91 @@ var (
 	logLevel = flag.String("loglevel", "Debug", "valid values are Debug, Info, Warning, Error")
 )
 
+type azureStorageClient struct {
+	accounts storage.AccountsClient
+	storage  azstorage.Client
+}
+
+func newAzureClients(ctx context.Context, m map[string]string) (*azureStorageClient, error) {
+	config := auth.NewClientCredentialsConfig(m["aadClientId"], m["aadClientSecret"], m["tenantId"])
+	authorizer, err := config.Authorizer()
+	if err != nil {
+		return nil, err
+	}
+
+	clients := &azureStorageClient{}
+	clients.accounts = storage.NewAccountsClient(m["subscriptionId"])
+	clients.accounts.Authorizer = authorizer
+
+	accts, err := clients.accounts.List(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	var acct storage.Account
+	var found bool
+	for _, acct = range *accts.Value {
+		found = acct.Tags["type"] != nil && *acct.Tags["type"] == "config"
+		if found {
+			break
+		}
+	}
+	if !found {
+		return nil, errors.New("storage account not found")
+	}
+	logrus.Printf("found account %s", *acct.Name)
+
+	keys, err := clients.accounts.ListKeys(context.Background(), m["resourceGroup"], *acct.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	clients.storage, err = azstorage.NewClient(*acct.Name, *(*keys.Keys)[0].Value, azstorage.DefaultBaseURL, azstorage.DefaultAPIVersion, true)
+	if err != nil {
+		return nil, err
+	}
+
+	return clients, nil
+}
+
 func sync() error {
 	logrus.Print("Sync process started")
 
-	b, err := ioutil.ReadFile("_data/containerservice.yaml")
+	b, err := ioutil.ReadFile("_data/_out/azure.conf")
+	if err != nil {
+		return errors.Wrap(err, "cannot read _data/_out/azure.conf")
+	}
+
+	var m map[string]string
+	if err := yaml.Unmarshal(b, &m); err != nil {
+		return errors.Wrap(err, "cannot unmarshal _data/_out/azure.conf")
+	}
+
+	az, err := newAzureClients(context.Background(), m)
```

Comment:
So we are running sync with `cloudConfig` credentials?

Reply:
yes- the cloudConfig credential is a service principal set up in the management tenant which has full rights over the resource group in the customer's subscription.  This is able to read the config data stored in the SA.